app/assets/main/react_components/helpers/AchievementsHelper.js

```diff
@@ -0,0 +1,192 @@
+import { uniqWith } from "lodash";
+
+const filterCategoryRelatedUserActions = (filter, condition, status) => {
+  return filter.filter(
+    (filterUserAction) =>
+      filterUserAction.climate_action_category_id === condition &&
+      filterUserAction.status === status
+  );
+};
+
+const filterCategoryRelatedActions = (filter, condition) => {
+  return filter.filter(
+    (filterUserAction) =>
+      filterUserAction.climate_action_category_id === condition
+  );
+};
+
+const findCategory = (performedColumn, item) => {
+  return performedColumn.find(
+    (performedItem) => item.climate_action_category_id == performedItem.id
+  );
+};
+
+const appendUserActionsArrayToCategory = (
+  climateActionCategories,
+  userActions
+) => {
+  return (
+    JSON.parse(climateActionCategories)
+      .map((category) => {
+        //The userActions with status true decides weather a categoryBadge should be created
+        const userPerformedActions = filterCategoryRelatedUserActions(
+          userActions,
+          category.id,
+          true
+        );
+        if (userPerformedActions.length !== 0) {
+          return {
+            ...category,
+            userActionsArray: userPerformedActions,
+          };
+        }
+      })
+      /** Filters the null values */
+      .filter((removed) => removed)
+  );
+};
+
+export const getCompleteCategoryArrays = (
+  actionsWithoutUserActions,
+  allUserActions,
+  climateActionCategories
+) => {
+  const userActions = allUserActions;
+  //Add actions
+  const res = appendUserActionsArrayToCategory(
+    climateActionCategories,
+    userActions
+  ).map((category) => {
+    const allActionsWithoutUserActions = filterCategoryRelatedActions(
+      JSON.parse(actionsWithoutUserActions),
+      category.id
+    );
+
+    //Add useractions with status false
+    const allUserActions = filterCategoryRelatedUserActions(
+      userActions,
+      category.id,
+      false
+    );
+
+    //map trough userActionsArray and save latest update timestamp and set to the whole category
+    return {
+      ...category,
+      userActionsArray: [...category.userActionsArray, ...allUserActions],
+      actionsArray: [...allActionsWithoutUserActions],
+      id: category.id.toString(),
+    };
+  });
+  return res;
+};
+
+export const updateAchievementsOnDrag = (
```

Comment:
Is this function only for Drag or is it for Move like one of the previous functions?